src/library/scala/collection/mutable/BufferLike.scala

```diff
@@ -105,15 +105,18 @@ trait BufferLike[A, +This <: BufferLike[A, This] with Buffer[A]]
     */
   def remove(n: Int): A
 
-  /** Removes a number of elements from a given index position.
+  /** Removes a number of elements from a given index position.  Subclasses of `BufferLike`
+   *  will typically override this method to provide better performance than `count`
+   *  successive calls to single-element `remove`.
    *
    *  @param n  the index which refers to the first element to remove.
    *  @param count  the number of elements to remove.
    *  @throws   IndexOutOfBoundsException if the index `n` is not in the valid range
-   *            `0 <= n <= length - count`.
+   *            `0 <= n <= length - count` (with `count > 0`).
```

Comment:
Ha. I guess that makes sense in a, if a non-tree didn't fall in a non-forest, did nobody not hear it kind of way...

You could make a similar argument that `"foo".substring(10, 10)` ought to return the empty string, since I didn't actually ask for any characters from the non-existent location, so hey man, no foul, here's the no characters you asked for. But it's a StringIndexOutOfBoundsException.

`(new java.lang.StringBuffer).delete(10, 10)`, same: StringIndexOutOfBoundsException. (But that isn't _exactly_ the same as yours, since it's two indices.)

I don't know what I think about this. You seem to know what you want here, so I guess I don't object. I'm certainly not volunteering to the audit the whole collections API to see whether we have a consistent position on this :-)